### PR TITLE
Allow the descendant fetcher to be more flexible by grouping uris by model

### PR DESCRIPTION
**Pull request by cjcolvar**
This PR preserves previous functionality for grouping uris by priority/other but also adds the capability to get the uris grouped by model if you call `descendant_and_self_uris_partitioned_by_model` directly.

### CODE REVIEW

File: lib/active_fedora/indexing/descendant_fetcher.rb [hunk 1]
```diff
@@ -90,10 +96,9 @@ def prioritized_object?
         end
 
         def add_self_to_partitioned_uris
-          if prioritized_object?
-            partitioned_uris[:priority] << rdf_resource.subject
-          else
-            partitioned_uris[:other] << rdf_resource.subject
+          rdf_graph_models.each do |model|
```
Comment (jrochkind) [inline lib/active_fedora/indexing/descendant_fetcher.rb]: If we're not using the `prioritized_object?` method anymore, should it be deleted?

**Comment by jrochkind** [comment]
Is this new behavior API-compatible with old behavior, people setting "priority models" can still set them in the old way and get the same behavior? I _think_ so, can you confirm?

**Comment by jrochkind** [comment]
@cjcolvar could it use any more docs explaining this use of `descendant_and_self_uris_partitioned_by_model`, and what it's intended for?

File: lib/active_fedora/indexing/descendant_fetcher.rb (hunk 1 above)
Comment (cjcolvar) [inline lib/active_fedora/indexing/descendant_fetcher.rb]: Good catch, I'll take it out.

**Comment by cjcolvar** [comment]
@jrochkind yes, I was intending to make it backwards compatible.  I didn't touch the tests and they still pass 
As for documentation, I was wondering if I should add more.  I'll do that and add some tests as well.

**Comment by jrochkind** [comment]
Awesome, thanks @cjcolvar, appreciate it!

**Comment by cjcolvar** [comment]
@jrochkind This is ready for rereview now.

**Comment by atz** [comment]
@cjcolvar Looks like you already have approval and just need a rebase

**Pull request merged**